app-modules/campaign/src/Actions/ExecuteCampaigns.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Assist\Campaign\Actions;
+
+use Assist\Campaign\Models\Campaign;
+use Assist\Campaign\Models\CampaignAction;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class ExecuteCampaigns implements ShouldQueue
+{
+    public function handle(): void
+    {
+        Campaign::query()
+            ->where('execute_at', '<=', now())
+            ->hasNotBeenExecuted()
+            ->cursor()
+            ->each(function (Campaign $campaign) {
+                $campaign->actions()->each(function (CampaignAction $action) {
+                    $action->execute();
+                });
+            });
```

Comment:
Could it be more fault-tolerant to make the execution of the campaign actions into a Job and have the command schedule a job per campaign?

Reply:
Or wait, is this done within `executeFromCampaignAction`?

Reply:
It is done within `executeFromCampaignAction`, currently not independently queued from this action. I will make that adjustment.